sphinx_immaterial/apidoc_formatting.py

```diff
@@ -200,17 +334,16 @@ def setup(app: sphinx.application.Sphinx):
     # to apply.
     sphinx.addnodes.desc_signature.classes.append("highlight")
 
-    app.add_config_value("html_wrap_signatures_with_css", default=None, rebuild="env")
+    app.connect("object-description-transform", _wrap_signatures)
+    _monkey_patch_object_description_to_include_fields_in_toc()
+
     app.add_config_value(
-        "html_wrap_signatures_with_css_column_limit", default=68, rebuild="env"
+        "object_description_options_map", default=[], rebuild="env", types=(dict,)
```

Comment:
Would it make sense to ease usability if we create a class(es) that can be used as user input to avoid erroneous/unsupported values? The revised config options seem a bit more complex (in terms of user input) than they were.

Reply:
I know this is an idea for a different PR, and I'm still trying wrap my head around the config changes. It looks like `object_description_options_patterns` can span several domains, and the user won't get notified if they typo'd an option in the build log. Sphinx intuitively lets the user know about an unrecognized/unused option in `html_theme_options`, but this intuition won't be extended to `object_description_options_patterns` or `object_description_options_map`.

Reply:
That is a good point.   Right now there is no registry of object description options, but we could add e.g. a `add_object_description_option` function to register them, and then could validate them like sphinx does for the config options.

Reply:
I added a registry and added type validation using pydantic.

Reply:
I'm now getting an error (repeated for each time I set `generate_synopses=None`):
```
ERROR: Invalid value None for object description option 'generate_synopses' specified for pattern re.compile('cpp:functionParam'): 1 validation error for ParsingModel[Literal[None, 'first_paragraph', 'first_sentence']]  
__root__
  none is not an allowed value (type=type_error.none.not_allowed)
```
 when using the config:
```py
object_description_options = [
    ("cpp:functionParam", dict(include_in_toc=False, generate_synopses=None)),
    ("cpp:function", dict(include_fields_in_toc=False, generate_synopses=None)),
]
```

Reply:
It seems like there may be a bug in pydantic.  In any case I changed the type to `Optional[Literal["first_sentence", "first_paragraph"]]` as a workaround, and that seems to solve the issue.